src/admin/resolvers/mutations/ApprovedItem.ts

```diff
@@ -144,6 +145,25 @@ export async function updateApprovedItem(
     );
   }
 
+  // To be able to delete authors associated with a corpus item, we first need
+  // to get the internal (integer) id for the story. This means doing a DB query
+  // to fetch the entire object.
+  const existingItem = await getApprovedItemByExternalId(
```

Comment:
Doesn't the association table have `externalId`?

Reply:
oh nvm... I think from reading the schema, that's another "external" ID for that specific author-item relationship?

I don't know prisma, but I feel like you *must* be able to delete by association... or add the additional FK

Reply:
The way this API is built (and also `collection-api`), with extra "external" ids to avoid exposing internal integer ones, means we can't actually delete related records by association until we get the full approved item record from the database that contains the integer primary key. 

It's the admin/internal part of the API so the performance hit of fetching this record an extra time is negligible. I would do it differently if performance was important.